commands/push.js

```diff
@@ -1,77 +1,68 @@
-'use strict';
+const cli = require('heroku-cli-util')
+const co = require('co')
+let Sanbashi = require('../lib/sanbashi')
 
-const cli = require('heroku-cli-util');
-const co = require('co');
-const child = require('child_process');
-
-module.exports = function(topic) {
+module.exports = function (topic) {
   return {
     topic: topic,
     command: 'push',
     description: 'Builds, then pushes a Docker image to deploy your Heroku app',
     needsApp: true,
     needsAuth: true,
-    args: [{ name: 'process', optional: true }],
-    flags: [{ name: 'verbose', char: 'v', hasValue: false }],
+    variableArgs: true,
+    flags: [
+      {
+        name: 'verbose',
+        char: 'v',
+        hasValue: false
+      },
+      {
+        name: 'recursive',
+        char: 'R',
+        hasValue: false
+      }
+    ],
     run: cli.command(co.wrap(push))
-  };
-};
+  }
+}
 
-function* push(context, heroku) {
-  let herokuHost = process.env.HEROKU_HOST || 'heroku.com';
-  let registry = `registry.${ herokuHost }`;
-  let proc = context.args.process || 'web';
-  let resource = `${ registry }/${ context.app }/${ proc }`;
+function* push (context, heroku) {
+  const recurse = !!context.flags.recursive
+  if(context.args.length === 0 && !recurse){
+    cli.error('Error: Requires either --recursive or one or more process types')
+    process.exit(1)
+  }
+  let herokuHost = process.env.HEROKU_HOST || 'heroku.com'
+  let registry = `registry.${ herokuHost }`
+  let dockerfiles = Sanbashi.getDockerfiles(process.cwd(), recurse)
+  let possibleJobs = Sanbashi.getJobs(`${ registry }/${ context.app }`, context.args, dockerfiles)
```

Comment:
I don't think this should use `path.join`. This is not a path even if it looks like one.